### PR TITLE
Background image fitting content 

### LINKED ISSUE #455: Soft Picks section for mobile
This section of the `soft-picks-advanced` page does not want to behave properly at some resolutions for mobile

https://main--sunstar--hlxsites.hlx.page/healthy-thinking/soft-picks-advanced1

Examples:

- At `900px` the section has a lot of empty space at the bottom because the background picture is really tall
Screenshot 2023-11-28 at 10 33 35
Screenshot 2023-11-28 at 10 34 51

- At `320px` the section text overflows
Screenshot 2023-11-28 at 10 35 20

- Other resolutions are a mix of the two, sometimes it looks just fine

@sdmcraft thoughts on how to tackle this?
cc @ehrro

### PR DESCRIPTION
**Pull request by dnbute**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after), along with a short summary of changes:

## Issue

Fixes #455 

## Changelog:

- Used container queries to sort of fix the issue of the image not resizing properly based on content

## Test URLs:
- Original: https://www.sunstar.com/healthy-thinking/soft-picks-advanced1
- Before: https://main--sunstar--hlxsites.hlx.live/healthy-thinking/soft-picks-advanced1
- After: https://i-455--sunstar--hlxsites.hlx.live/healthy-thinking/soft-picks-advanced1

## Library

- [ ] New Blocks introduced in this PR
      If yes, please provide details below
**Block Name** : For e.g. _cards_
- [ ] Documented in sidekick library

- [ ] New variations introduced in this PR
**Variation Name** :  For e.g. _cards (grid)_
- [ ] Documented in sidekick library

- [ ] New mixins introduced in this PR
**Mixin Name** :  For e.g. _compact, white_
- [ ] Documented in sidekick library

### CODE REVIEW

**Comment by dnbute** [comment]
@sdmcraft This sort of fixes the image not being the right size for its content, but with 2 pretty big caveats
1. This won't work on firefox since I use `:has`, I can change some things in the js so I have a specific class I can select, so this is fixable
2. This is a bit esoteric, I'm not sure how much support there is for container queries (intellisense doesnt even know about them), and I remember I used them at one point the past as well and Dejan got rid of them

**Comment by sdmcraft** [comment]
@dnbute , I like the idea of using `containers` for achieving this. Yes if we can do it without `:has` , it'd be great. I guess all other features used in here like `containers` and `cqh` are widely supported.
/cc @ehrro 
[0] - https://caniuse.com/css-container-queries
[1] - https://caniuse.com/css-container-query-units

**Comment by ehrro** [comment]
Hello @dnbute thanks for this actually I wanted to ask for this update.

I agree with @sdmcraft regarding the :has since it's not yet supported by all the browser I would like to avoid using it.

Thanks.

**Comment by dnbute** [comment]
@sdmcraft @ehrro, so, this is sort of the best I could do, I can fine-tune the value some more but if I make it so it works at lower resolutions, it's bad on higher resolutions and vice-versa

Also, this is a kinda weird use-case we have, cause I'm updating the image based on its container, container which has its height set in part by the image, so there's some circularity going on.

Also also this is only for mobile

File: styles/styles.css [hunk 1]
```diff
@@ -152,18 +152,17 @@
...
     container-name: text-container;
     container-type: inline-size;
   }
   
   /* this updates the height of the content inside the section */
-  @container text-container (max-width: 62rem) {
-    body > main > div.with-background-image > picture img,
-    body > main > div.with-background-image > .section-container {
-      height: 110cqh; /* 110% of the container's height */
+  @container text-container (max-width: 991px) {
```
Comment (sdmcraft) [inline styles/styles.css:161]: By convention, we are following `min-width`. Could you please adjust for the same? Basically make this the default style. BTW, does it make sense to have this apply to wide screens as well?

**Comment by sdmcraft** [comment]
@dnbute , are you planning to merge this? I think this is useful and should be good to merge once you reviewed the comments.
/cc @ehrro

**Comment by dnbute** [comment]
@sdmcraft I'm not sure this actually works for all use cases and I'm afraid it might break more stuff than it fixes; I'd have to test this more before I try to merge it

**Comment by dnbute** [comment]
But something broke on main it seems

Screenshot 2023-12-05 at 10 27 30

**Comment by sdmcraft** [comment]
@dnbute , should we bookmark this PR in the linked issue and close this if you aren't planning to take it up anytime soon?

**Comment by dnbute** [comment]
@sdmcraft Yeah sure, unfortunately it seems this doesn't want to behave the way I want it to, and since this is not a high priority change then yeah, we can close it and maybe take it up another time

**Pull request closed without merging**